### ISSUE TITLE
[RFQ] Add check translation-not-lazy: Prefer _("%s", var) instead of _("%s" % var) similar to logging-not-lazy

### ISSUE THREAD

**Issue by moylop260** [OCA/pylint-odoo#378]
The pylint check `logging-not-lazy` shows the following message:
 - `Use lazy % or .format() formatting in logging functions`

For the following bad code:

```python
_logger.info("%s" % var)
```

The good one:

```python
_logger.info("%s", var)
```

A similar check is needed for the translation method `odoo.tools.translate._`

To say `Use lazy % or .format() formatting in translation functions`

For the following bad code:

```python
_("%s" % var)
```

The good one:

```python
_("%s", var)
```

Notice that this check only will be raised for odoo >=14.0

<=13.0 does not have positional arguments:
 - https://github.com/odoo/odoo/blob/a03aa23461498bba67/odoo/tools/translate.py#L447

>=14.0 has positional arguments:
 - https://github.com/odoo/odoo/blob/6a6e5f92789e1c/odoo/tools/translate.py#L446

So, we only are limiting this check for odoo >=14.0 since past version is not already valid

Using a similar way than:
 - https://github.com/OCA/pylint-odoo/pull/377/files

It is because of the old way is [blocking the execution of the code, fallback on source
string.](https://github.com/odoo/odoo/commit/3c3b88f6768e31417263b3828a874ca204e29514)

**Comment by moylop260** [comment]
@antonag32 please

**Comment by luisg123v** [comment]
:+1: 

CC @frahikLV

**Comment by moylop260** [comment]
I just realized all the [logging](https://github.com/PyCQA/pylint/blob/7d4569ed77c6b16f1f1eb623887ef221e6784b0c/pylint/checkers/logging.py#L131) checks apply for `odoo._` translation method in the same way. e.g. logging-too-many-args, logging-not-lazy...

We could use the extra parameter `logging-modules` but it is even don't work because of it needs to change the global variable [CHECKED_CONVENIENCE_FUNCTIONS](https://github.com/PyCQA/pylint/blob/7d4569ed77c6b16f1f1eb623887ef221e6784b0c/pylint/checkers/logging.py)


I just did a POC applying the following patch and it is working as expected:

```diff
diff --git a/pylint/checkers/logging.py b/pylint/checkers/logging.py
index 53ac24f60..7afc0a8b9 100644
--- a/pylint/checkers/logging.py
+++ b/pylint/checkers/logging.py
@@ -103,6 +103,7 @@ CHECKED_CONVENIENCE_FUNCTIONS = {
     "info",
     "warn",
     "warning",
+    "_",
 }


@@ -138,7 +139,7 @@ class LoggingChecker(checkers.BaseChecker):
         (
             "logging-modules",
             {
-                "default": ("logging",),
+                "default": ("logging", "odoo"),
                 "type": "csv",
                 "metavar": "<comma separated list>",
                 "help": "Logging modules to check that the string format "
@@ -215,7 +216,9 @@ class LoggingChecker(checkers.BaseChecker):
                         return True, inferred._proxied.name
             return False, None

-        if is_logging_name():
+        if "_(" in node.as_string():
+            name = "_"
+        elif is_logging_name():
             name = node.func.attrname
         else:
             result, name = is_logger_class()
```


We just need to define a similar way without patches using pylint-odoo
